### ISSUE TITLE
fix bug Orion with MongoDB 3.0 and 2.6 using Authentication

### ISSUE THREAD

**Issue by minhtri1990** [telefonicaid/fiware-orion#2987]
I thing Class connectionOperations.cpp at 
fiware-orion/src/lib/mongoBackend/connectionOperations.cpp
the code must change to.

```
try
  {
	//mongo db 3.0
	mongo::BSONObj params = BSON("mechanism" << "SCRAM-SHA-1" << "user" << user << "db" << db << "pwd" << password);	
	//mongo db 2.6
	//mongo::BSONObj params = BSON("mechanism" << "MONGODB-CR" << "user" << user << "db" << db << "pwd" << password);
    //connection->auth(params);	
	if (!connection->auth(params))
    {
      std::string msg = std::string("authentication fails: db=") + db +
          ", username='" + user + "'" +
          ", password='*****'" +
          ", auth_error='" + authErr + "'";
      *err = "Database Startup Error (" + msg + ")";
      LM_E((err->c_str()));
      return false;
    }
```

**Comment by fgalan** [comment]
Maybe related with issues https://github.com/telefonicaid/fiware-orion/issues/1061 and https://github.com/telefonicaid/fiware-orion/issues/2962. All three could be aspects of the same issue.

**Comment by osfrance** [comment]
Hi, I had the same issue when I try to connect Orion with MongoDB and authentification. It's an important problem for my organization, because we must put IOT infrastructure in production and we can't use mongoDB without authentification mechanism. Thanks for reply.

**Comment by fgalan** [comment]
Did you tried the build from source procedure, compiling MongoDB driver with SASL and SSL support https://github.com/telefonicaid/fiware-orion/blob/master/doc/manuals/admin/build_source.md#building-mongodb-driver-with-sasl-and-ssl-support ?

**Comment by fgalan** [comment]
PR https://github.com/telefonicaid/fiware-orion/pull/3544 implement the capability of changing the authentication mechanism. In addition, it sets SCRAM-SHA-1 as default.

@mathi123 you told "Those instructions do not work". Do you mean my comment at https://github.com/telefonicaid/fiware-orion/issues/2987#issuecomment-378037504? Or the modifications proposed by @minhtri1990 (which are the ones used in PR #3544)?

**Comment by fgalan** [comment]
I think this is fixed by PR #3544 and #3542. Thus, I'm closing this issue.

If I'm wrong, please @mathi123 @minhtri1990 tell me so we can reopen on create a new one. Thanks!
